src/NuGet.Core/NuGet.Configuration/PackageNamespaces/SearchTree.cs

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace NuGet.Configuration
+{
+    internal class SearchTree
```

Comment:
I don't really have a strong preference of generic vs dedicated type, especially because this is internal. 

If necessary, this can be a made generic at a later point. This might be a case of premature optimization.

Reply:
I have no objection, since currently this class is internal type we can do it.

Reply:
Not important for now.